applications/GeoMechanicsApplication/custom_retention/van_genuchten_law.cpp

```diff
@@ -29,17 +26,17 @@ double VanGenuchtenLaw::CalculateSaturation(Parameters& rParameters) const
 {
     KRATOS_TRY
 
-    const double&     p                   = rParameters.GetFluidPressure();
-    const Properties& rMaterialProperties = rParameters.GetMaterialProperties();
+    const auto  p                   = rParameters.GetFluidPressure();
+    const auto& rMaterialProperties = rParameters.GetMaterialProperties();
 
     if (p > 0.0) {
-        const double& satMax = rMaterialProperties[SATURATED_SATURATION];
-        const double& satMin = rMaterialProperties[RESIDUAL_SATURATION];
-        const double& pb     = rMaterialProperties[VAN_GENUCHTEN_AIR_ENTRY_PRESSURE];
-        const double& gn     = rMaterialProperties[VAN_GENUCHTEN_GN];
-        const double  gc     = (1.0 - gn) / gn;
+        const auto sat_max = rMaterialProperties[SATURATED_SATURATION];
+        const auto sat_min = rMaterialProperties[RESIDUAL_SATURATION];
+        const auto pb      = rMaterialProperties[VAN_GENUCHTEN_AIR_ENTRY_PRESSURE];
+        const auto gn      = rMaterialProperties[VAN_GENUCHTEN_GN];
+        const auto gc      = (1.0 - gn) / gn;
 
-        return satMin + (satMax - satMin) * pow(1.0 + pow(p / pb, gn), gc);
+        return sat_min + (sat_max - sat_min) * pow(1.0 + pow(p / pb, gn), gc);
```

Comment:
I would suggest to use `std::pow` here:
```suggestion
        return sat_min + (sat_max - sat_min) * std::pow(1.0 + std::pow(p / pb, gn), gc);
```

There are more occurrences of `pow` in this file that you may want to replace by `std::pow`.

Reply:
replaced all.